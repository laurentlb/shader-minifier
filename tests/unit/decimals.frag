//NOCOMPILE
void test () {
  x(3.141592653589793);
  x(3.1415927);
  //x(1e308);
  //x(3e38);
  //x(1e37);
  x(1e28);
  x(125.663704/180.);
  x(.6981316889);
  x(100000005.);
  x(1.24e27);
  x(1.20e27);
  x(1.00e27);
  x(1.24e4);
  x(1.20e4);
  x(1.00e4);
  x(1.24e3);
  x(1.20e3);
  x(1.00e3);
  x(1.24e2);
  x(1.20e2);
  x(1.00e2);
  x(1.24e1);
  x(1.20e1);
  x(1.00e1);
  x(1.24e0);
  x(1.20e0);
  x(1.00e0);
  x(1.24e-1);
  x(1.20e-1);
  x(1.00e-1);
  x(1.24e-2);
  x(1.20e-2);
  x(1.00e-2);
  x(1.24e-3);
  x(1.20e-3);
  x(1.00e-3);
  x(1.24e-4);
  x(1.20e-4);
  x(1.00e-4);
  x(1.24e-27);
  x(1.20e-27);
  x(1.00e-27);
  x(6e-29);
  x(8e-46);
  //x(3e-324);
}
