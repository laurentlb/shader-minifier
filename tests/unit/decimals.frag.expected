void test()
{
  x(acos(-1.));
  x(3.1415927);
  x(1e28);
  x(125.663704/180.);
  x(.6981316889);
  x(100000005.);
  x(124e25);
  x(12e26);
  x(1e27);
  x(124e2);
  x(12e3);
  x(1e4);
  x(1240.);
  x(12e2);
  x(1e3);
  x(124.);
  x(120.);
  x(1e2);
  x(12.4);
  x(12.);
  x(10.);
  x(1.24);
  x(1.2);
  x(1.);
  x(.124);
  x(.12);
  x(.1);
  x(.0124);
  x(.012);
  x(.01);
  x(.00124);
  x(.0012);
  x(.001);
  x(124e-6);
  x(12e-5);
  x(1e-4);
  x(0.);
  x(0.);
  x(0.);
  x(0.);
  x(0.);
}
